src/conways.rs

```diff
@@ -0,0 +1,126 @@
+const COLUMNS: usize = 32;
+const ROWS: usize = 32;
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum CellState {
+    Alive,
+    Dead,
+}
+
+#[derive(Clone, Copy)]
+pub struct Position(pub usize, pub usize);
+
+pub struct Conways {
+    pub grid: Vec<Vec<CellState>>,
+}
+
+impl Conways {
+
+    // Using this function in the test
+    #[allow(dead_code)]
+    pub fn default() -> Self {
+        Self {
+            grid: vec![vec![CellState::Dead; COLUMNS]; ROWS]
+        }
+    }
+
+    pub fn from(alive_cells: Vec<Position>) -> Self {
+        let mut grid = vec![vec![CellState::Dead; COLUMNS]; ROWS];
+        
+        for Position(x, y) in alive_cells {
+            if x < ROWS && y < COLUMNS {
+                grid[x][y] = CellState::Alive;
+            }
+        }
+        
+        Self { grid }
+    }
+
+    pub fn update_cells(&mut self) {
+        let mut new_grid = self.grid.clone();
+    
+        for (x, _) in self.grid.iter().enumerate() {
+            for (y, _) in self.grid.iter().enumerate() {
+                let neighbors = self.count_neighbors(Position(x, y));
+                new_grid[x][y] = match (self.grid[x][y], neighbors) {
+                    (CellState::Alive, 2 | 3) => CellState::Alive,
+                    (CellState::Alive, _) => CellState::Dead,
+                    (CellState::Dead, 3) => CellState::Alive,
+                    (CellState::Dead, _) => CellState::Dead,
+                };
+            }
+        }
+    
+        self.grid = new_grid;
+    }
+
+    pub fn count_neighbors(&self, position: Position) -> usize {
+        let mut count = 0;
+
+        // Position of the eight neighbors
+        let directions = [
+            (-1, -1), (-1, 0), (-1, 1),
+            (0, -1),           (0, 1),
+            (1, -1),  (1, 0),   (1, 1),
+        ];
+
+        for (x, y) in directions.iter() {
+            let new_x = position.0 as isize + x;
+            let new_y = position.1 as isize + y;
+
+            // Avoid leaving the grid
+            if new_x >= 0 && new_x < ROWS as isize && new_y >= 0 && new_y < COLUMNS as isize {
+                if let CellState::Alive = self.grid[new_x as usize][new_y as usize] {
```

Comment:
You don't need to assign anything with let here, I think you can just do
`if self.grid[new_x as usize][new_y as usize] == CellState::Alive `

Reply:
You're right. I will change it.